### PR TITLE
Switch AOT Inductor test to export, add dynamic, fix invocation bug

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #101585

Fixes

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by ezyang** [comment]
I'm unfamiliar with the code modified here so I don't know how to review this without more information.

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -1646,13 +1654,22 @@ def call_kernel(self, code, name: str):
...
-                code.writeline(f"{expr} = {pexpr(tree.numel)}")
+                # TODO(voz): Tragic. This should at the very least be a util to slapp on declare and ending.
+                # The real fix here is to revisit our cross language calling convention.
+                code.writeline(
+                    f"{code.declare}{expr} = {pexpr(tree.numel)}{code.ending}"
+                )
             if tree.prefix != "r" or self.inside_reduction:
-                call_args.append(expr)
+                if assignment:
+                    call_args.append(SymbolicCallArg(expr))
```
Comment (ngimel) [inline torch/_inductor/codegen/triton.py:1670]: why is it `str` and not `pexpr`?

File: torch/_inductor/codegen/triton.py [hunk 2]
```diff
@@ -1662,7 +1679,7 @@ def call_kernel(self, code, name: str):
             )
         else:
             # TODO: refactor generate_kernel_call
-            call_args_str = ", ".join(call_args)
+            call_args_str = ", ".join(str(item) for item in call_args)
```
Comment (ngimel) [inline torch/_inductor/codegen/triton.py:1682]: `str` here because args can be symbolic expressions? Can you leave a comment?

**Review by ngimel — APPROVED** [review]

**Review by desertfire — APPROVED** [review]
If you want to fix more dynamic shape errors on aot inductor, you can enable it for https://github.com/pytorch/pytorch/blob/main/test/inductor/test_cpp_wrapper.py and fix them. Thanks.

**Comment by voznesenskym** [comment]
> If you want to fix more dynamic shape errors on aot inductor, you can enable it for https://github.com/pytorch/pytorch/blob/main/test/inductor/test_cpp_wrapper.py and fix them. Thanks.

https://github.com/pytorch/pytorch/pull/100815 drives this, for all of PT2, when I have sound pieces to break off (like this) I make PRs from them (like this).

File: torch/_inductor/codegen/triton.py (hunk 2 above)
Comment (voznesenskym) [reply torch/_inductor/codegen/triton.py:1682]: Yes, will add a comment.

File: torch/_inductor/codegen/triton.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_inductor/codegen/triton.py:1670]: Sorry, what do you mean? Why do we call `str()` on item?

**Pull request closed without merging**